lib/ahoy_email/processor.rb

```diff
@@ -68,20 +68,14 @@ def options
     end
 
     def generate_token
-      SecureRandom.urlsafe_base64(32).gsub(/[\-_]/, "").first(32)
+      @token ||= SecureRandom.urlsafe_base64(32).gsub(/[\-_]/, "").first(32)
     end
 
     def track_open
       if html_part?
         raw_source = (message.html_part || message).body.raw_source
         regex = /<\/body>/i
-        url =
-          url_for(
-            controller: "ahoy/messages",
-            action: "open",
-            id: ahoy_message.token,
-            format: "gif"
-          )
+        url = default_url.gsub("$ID$", generate_token).gsub("/click", "/open.gif")
```

Comment:
Should we use `sub` instead of `gsub`? Also, should probably anchor the click -> open substitution just to be safe.